examples/adjoints/adjoints_ex5/adjoints_ex5.C

```diff
@@ -578,9 +581,10 @@ int main (int argc, char ** argv)
       // Now that the adjoint initial condition is set, we will start the
       // backwards in time adjoint integration
 
-      // For loop stepping backwards in time
-      for (unsigned int t_step=param.initial_timestep;
-           t_step != param.initial_timestep + param.n_timesteps; ++t_step)
+      // A backwards in time while loop
+      unsigned int t_step = 0;
+
+      while(system.time > 0)
```

Comment:
This again seems to hardcode a special t=0.  Not nearly as bad in an app as in the library code ... but it does seem to break the initial_timestep restart support here, which I think ought to be workable even if it's not nearly as useful.

Reply:
If we use a fixed time step solver, we have n timesteps, with the two step solver, we have 2*n timesteps. I didnt want to add an if conditional, but I guess we are already doing that to check the gold values.

Reply:
But we don't have 2*n timesteps.  We have n timesteps, with 2*n sub-timesteps.  TwoStepTimeSolver::adjoint_solve should be handling the 2 core_time_solver->adjoint_solve() calls, the same way TwoStepTimeSolver::solve handles 2 core_time_solver->solve() calls...

Reply:
This would mean changes to the adjoint_solve() method of system, it will have to call a different adjoint_solve() implementation depending on the type of timesolver we are using. Do we want those changes to be part of this PR ?

Reply:
It wouldn't be any change to ImplicitSystem::adjoint_solve(), it would be a change to DifferentiableSystem::adjoint_solve() and an addition of TimeSolver::adjoint_solve() .. which in hindsight we should have had years ago.  Instead of just setting a flag before handing off to ImplicitSystem, DifferentiableSystem would be calling time_solver->adjoint_solve(), analogous to what it does in the solve() case.  Some of the TimeSolver subclasses would still be able to just set a flag, but things like TwoStep would actually do two steps.